src/session_impl.cpp

```diff
@@ -3145,7 +3133,8 @@ namespace aux {
 			for (auto& ext : m_ses_extensions)
```

Comment:
I was thinking about a map, because having multiple lists is a nightmare to maintain, but I feared throwing to much data structure could impact performance in each iteration.

If you don't mind, how the "last minute check" is an anti-pattern? just to know your opinion in the matter.